plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -77,6 +136,102 @@ protected String getStorageMetadataLocation(String materializedViewName)
+        assertThat(materializedRows.get(0).getField(1)).isEqualTo(2);
+        int valueFromPtf2 = (int) materializedRows.get(0).getField(0);
+        assertThat(valueFromPtf2).isNotEqualTo(valueFromPtf1); // differs because PTF sequence_function is called directly as mv is considered stale
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("STALE");
+        assertThat(computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isNull();
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        assertThat((String) computeActual("SELECT freshness FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue()).isEqualTo("UNKNOWN");
+        ZonedDateTime lastFreshTime = (ZonedDateTime) computeActual("SELECT last_fresh_time FROM system.metadata.materialized_views WHERE name = '" + viewName + "'").getOnlyValue();
+        assertThat(lastFreshTime).isNotNull();
+        materializedRows = computeActual("SELECT * FROM " + viewName + " ORDER BY next_value").getMaterializedRows();
+        assertThat(materializedRows.size()).isEqualTo(1);
+        assertThat(materializedRows.get(0).getField(1)).isEqualTo(2);
+        int valueFromPtf3 = (int) materializedRows.get(0).getField(0);
+        assertThat(valueFromPtf3).isNotEqualTo(valueFromPtf1);
+        assertThat(valueFromPtf3).isNotEqualTo(valueFromPtf2);
+
+        materializedRows = computeActual("SELECT * FROM " + viewName + " ORDER BY next_value").getMaterializedRows();
+        int valueFromPtf4 = (int) materializedRows.get(0).getField(0);
+        assertThat(materializedRows.size()).isEqualTo(1);
+        assertThat(materializedRows.get(0).getField(1)).isEqualTo(2);
+        assertThat(valueFromPtf4).isNotEqualTo(valueFromPtf1);
+        assertThat(valueFromPtf4).isNotEqualTo(valueFromPtf2);
+        assertThat(valueFromPtf4).isEqualTo(valueFromPtf3); // mv is not stale anymore so all selects until next refresh returns same result
+    }
+
+    @Test
+    public void testMaterializedViewCreatedFromPtfWithGracePeriod()
+            throws InterruptedException
+    {
+        String viewName = "materialized_view_for_ptf_with_grace_period_" + randomNameSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " GRACE PERIOD INTERVAL '1' SECOND AS SELECT * FROM TABLE(mock.system.sequence_function())");
+
+        int result1 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        int result2 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result2).isNotEqualTo(result1);
+
+        assertUpdate("REFRESH MATERIALIZED VIEW " + viewName, 1);
+        int result3 = (int) computeActual("SELECT * FROM " + viewName).getOnlyValue();
+        assertThat(result3).isNotEqualTo(result2);
```

Comment:
One way to check that the last result is different from all the previous results
```
assertThat(List.of(result1, result2)).doesNotContain(result3);
```